pkg/apiserver/api_installer.go

```diff
@@ -110,62 +110,54 @@ func (a *APIInstaller) NewWebService() *restful.WebService {
 	return ws
 }
 
-func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storage, ws *restful.WebService, proxyHandler http.Handler) (*unversioned.APIResource, error) {
-	admit := a.group.Admit
-	context := a.group.Context
-
-	optionsExternalVersion := a.group.GroupVersion
-	if a.group.OptionsExternalVersion != nil {
-		optionsExternalVersion = *a.group.OptionsExternalVersion
-	}
-
-	var resource, subresource string
-	switch parts := strings.Split(path, "/"); len(parts) {
-	case 2:
-		resource, subresource = parts[0], parts[1]
-	case 1:
-		resource = parts[0]
-	default:
-		// TODO: support deeper paths
-		return nil, fmt.Errorf("api_installer allows only one or two segment paths (resource or resource/subresource)")
+// validFQKind validates and returns the group version kind registered for the given storage
+// object. If the storage object is a subresource and has an override specified, the object's
+// group version is validated against the group version in the override for that subresource.
+func (a *APIInstaller) validFQKind(path string, storage rest.Storage) (unversioned.GroupVersionKind, error) {
+	fqKindToRegister := unversioned.GroupVersionKind{}
+	allowedGroup := a.group.GroupVersion.Group
+	if override, ok := a.group.SubresourceOverrides[path]; ok {
+		allowedGroup = override.GroupVersion.Group
 	}
-	hasSubresource := len(subresource) > 0
 
 	object := storage.New()
 	fqKinds, err := a.group.Typer.ObjectKinds(object)
 	if err != nil {
-		return nil, err
+		return fqKindToRegister, err
```

Comment:
This is just an empty `unversioned.GroupVersionKind`?  Please return that explictly to make it clear the value is meant to be invalid/unused.